### PR TITLE
Feat: support of multiple selection in media upload

### PR DESCRIPTION
**Pull request by dratwas**
In this PR I added support for multiple selections of media in MediaUpload component.  It is a breaking change and apps will crash without changes from listed PRs.

Associated issue - https://github.com/wordpress-mobile/gutenberg-mobile/issues/1298

Related gutenberg PR - https://github.com/WordPress/gutenberg/pull/17397
Related Wordpres-iOS PR - https://github.com/wordpress-mobile/WordPress-iOS/pull/12404
Related Wordpress-Android PR - https://github.com/wordpress-mobile/WordPress-Android/pull/10556
To test:
- set multiple prop to true in image/video block
- run Wordpress-iOS or Wordpress-Android
- add image/video block - multiple select should be enabled for selecting media from the device or WP Media
- the array of selected media should be passed to the callback function

NOTE: Multiple selections of media from the device on iOS doesn't work since this functionality is implemented by the client app. Our example app from `ios` directory doesn't support multiple selections from the device because it uses build-in `UIImagePickerController`. In order to make it works in our example app, I would need to add an external library that supports multiple selections.

### CODE REVIEW

File: react-native-gutenberg-bridge/ios/GutenbergBridgeDelegate.swift [hunk 1]
```diff
@@ -63,7 +63,7 @@ public protocol GutenbergBridgeDelegate: class {
     ///     - source: the source from where the picker will get the media
     ///     - callback: A callback block to be called with an upload mediaIdentifier and a placeholder image file url, use nil on both parameters to signal that the action was canceled.
```
Comment (pinarol) [inline react-native-gutenberg-bridge/ios/GutenbergBridgeDelegate.swift]: I know this is still draft but let's not forget to update `callback` definition also.

File: react-native-gutenberg-bridge/ios/RNReactNativeGutenbergBridge.swift [hunk 2]
```diff
@@ -22,12 +22,23 @@ public class RNReactNativeGutenbergBridge: RCTEventEmitter {
...
+                    let formattedMedia = media.map { (id, url) in
+                        return ["id": id, "url": url]
+                    }
+                    callback([formattedMedia])
+                } else {
+                    guard let (mediaID, mediaURL) = media.first else {
+                        callback(nil)
+                        return
+                    }
+                    callback([["id": mediaID, "url": mediaURL]])
```
Comment (pinarol) [inline react-native-gutenberg-bridge/ios/RNReactNativeGutenbergBridge.swift]: we can define constants for "id" and "url"

File: react-native-gutenberg-bridge/ios/RNReactNativeGutenbergBridge.swift [hunk 3]
```diff
@@ -39,12 +50,12 @@ public class RNReactNativeGutenbergBridge: RCTEventEmitter {
             return
         }
         DispatchQueue.main.async {
-            self.delegate?.gutenbergDidRequestImport(from: url, with: { (mediaID, url) in
-                guard let url = url, let mediaID = mediaID else {
+            self.delegate?.gutenbergDidRequestImport(from: url, with: { media in
```
Comment (pinarol) [inline react-native-gutenberg-bridge/ios/RNReactNativeGutenbergBridge.swift]: I think `media` is an array now? so how about renaming to sth like `mediaList`?

**Comment by pinarol** [comment]
This is looking pretty good overall, I just wanted to point out some things early on.

So what are some pending work here?

**Comment by dratwas** [comment]
Prop `multiple` works on iOS with these changes so basically I need to check how the `cancelation` mechanism works and make it works with multiple as well and add tests.

**Comment by pinarol** [comment]
thanks @dratwas ! did you have chance to try multiple uploads also?

**Comment by pinarol** [comment]
> and add tests.

really appreciate that!

**Comment by dratwas** [comment]
> thanks @dratwas ! did you have chance to try multiple uploads also?

AFAIK this is a responsibility of the `client` app. I implemented it in `Wordpress-iOS` here https://github.com/wordpress-mobile/WordPress-iOS/pull/12404/files#diff-0992f0530230d5beb61d0569d50f4459R40
I tried it and native side uploads media and sends all "mediaUpload"  events related to it but it is not handled on JS side because at this moment JS implementation of `image/edit.native.js` assumes that we can get only one image. So it is prepared to support multiple media upload but needs to be handled in `gallery` block.

**Comment by pinarol** [comment]
> I tried it and native side uploads media and sends all "mediaUpload" events related to it 

That's actually enough for this particular task thanks! Sorry If I confused you, the rest will be handled in `gallery` block. I just wanted to make sure we are sending the "mediaUpload" events generated by simultaneous uploads. 👍

**Comment by pinarol** [comment]
Could you be able to update the branch from `develop` ?

**Comment by pinarol** [comment]
Could you also add the associated issue to PR description?

File: ios/gutenberg/GutenbergViewController.swift [hunk 4]
```diff
@@ -70,9 +70,9 @@ extension GutenbergViewController: GutenbergBridgeDelegate {
             print("Gutenberg did request media picker, passing a sample url in callback")
             switch currentFilter {
             case .image:
-                callback(1, "https://cldup.com/cXyG__fTLN.jpg")
+                callback([(id: 1, url: "https://cldup.com/cXyG__fTLN.jpg")])
             case .video:
-                callback(2, "https://i.cloudup.com/YtZFJbuQCE.mov")
+                callback([(2, "https://i.cloudup.com/YtZFJbuQCE.mov")])
```
Comment (pinarol) [inline ios/gutenberg/GutenbergViewController.swift]: Looking inconsistent compared to line 73 since it has `id:`, `url:` labels for tuple items

**Comment by pinarol** [comment]
Could you be able to update the branch resolve merge conflicts one more time? Sorry for being late for reviewing, I'll start after this. Thanks!

**Comment by mkevins** [comment]
I tested this on Android, and it is working nicely. :tada:

File: .gitmodules [hunk 5]
```diff
@@ -1,6 +1,6 @@
 [submodule "gutenberg"]
 	path = gutenberg
-	url = ../../WordPress/gutenberg.git
+	url = https://github.com/callstack/gutenberg.git
```
Comment (SergioEstevao) [inline .gitmodules]: Should this stay with this value after the merge?

Comment (dratwas) [reply .gitmodules]: No, it should be reverted before the merge

**Pull request merged**